src/python/Pose3.i

```diff
@@ -0,0 +1,104 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+%module quaternion
+%{
+  #include <ignition/math/Pose3.hh>
+  #include <ignition/math/Quaternion.hh>
+  #include <ignition/math/Vector3.hh>
+  #include <ignition/math/config.hh>
+%}
+
+%include "std_string.i"
+%include "Quaternion.i"
+
+
+namespace ignition
+{
+  namespace math
+  {
+    template<typename T>
+    class Pose3
+    {
+      %rename("%(undercase)s", %$isfunction, %$ismember, %$not %$isconstructor) "";
+      %rename("%(uppercase)s", %$isstatic, %$isvariable) "";
+      
+      public: static const Pose3<T> Zero;
+
+      public: Pose3() : p(0, 0, 0), q(1, 0, 0, 0);
+      public: Pose3(const Vector3<T> &_pos, const Quaternion<T> &_rot)
+      : p(_pos), q(_rot);
+      public: Pose3(T _x, T _y, T _z, T _roll, T _pitch, T _yaw)
+      : p(_x, _y, _z), q(_roll, _pitch, _yaw);
+      public: Pose3(T _x, T _y, T _z, T _qw, T _qx, T _qy, T _qz)
+      : p(_x, _y, _z), q(_qw, _qx, _qy, _qz);
+      public: Pose3(const Pose3<T> &_pose)
+      : p(_pose.p), q(_pose.q);
+      public: virtual ~Pose3();
+      public: void Set(const Vector3<T> &_pos, const Quaternion<T> &_rot);
+      public: void Set(const Vector3<T> &_pos, const Vector3<T> &_rpy);
+      public: void Set(T _x, T _y, T _z, T _roll, T _pitch, T _yaw);
+      public: bool IsFinite() const;
+      public: inline void Correct();
+      public: Pose3<T> Inverse() const;
+      public: Pose3<T> operator+(const Pose3<T> &_pose) const;
+      public: const Pose3<T> &operator+=(const Pose3<T> &_pose);
+      public: inline Pose3<T> operator-() const;
+      public: inline Pose3<T> operator-(const Pose3<T> &_pose) const;
+      public: const Pose3<T> &operator-=(const Pose3<T> &_pose);
+      public: bool operator==(const Pose3<T> &_pose) const;
+      public: bool operator!=(const Pose3<T> &_pose) const;
+      public: Pose3<T> operator*(const Pose3<T> &_pose) const;
+      public: const Pose3<T> &operator*=(const Pose3<T> &_pose);
+      public: Vector3<T> CoordPositionAdd(const Vector3<T> &_pos) const;
+      public: Vector3<T> CoordPositionAdd(const Pose3<T> &_pose) const;
+      public: inline Vector3<T> CoordPositionSub(const Pose3<T> &_pose) const;
+      public: Quaternion<T> CoordRotationAdd(const Quaternion<T> &_rot) const;
+      public: inline Quaternion<T> CoordRotationSub(
+                  const Quaternion<T> &_rot) const;
+      public: Pose3<T> CoordPoseSolve(const Pose3<T> &_b) const;
+      public: void Reset();
+      public: Pose3<T> RotatePositionAboutOrigin(const Quaternion<T> &_q) const;
+      public: void Round(int _precision);
+      public: inline const Vector3<T> &Pos() const;
+      public: inline Vector3<T> &Pos();
```

Comment:
Are they differentiable in python?

Reply:
No, they are not. I addressed this and did the same with the `Rot()` method, that was the same case.